test/spmd/tensor/test_matrix_ops.py

```diff
@@ -127,6 +131,150 @@ def test_t(self):
+            batch_2_placements: Sequence[Placement],
+            beta: int,
+            alpha: int,
+        ) -> None:
+            tensor_dt = DTensor.from_local(tensor, device_mesh, tensor_placements)
+            batch_1_dt = DTensor.from_local(batch_1, device_mesh, batch_1_placements)
+            batch_2_dt = DTensor.from_local(batch_2, device_mesh, batch_2_placements)
+            new_dt = torch.baddbmm(
+                tensor_dt, batch_1_dt, batch_2_dt, beta=0.0, alpha=0.5
+            )
+            self.assertEqual(new_dt.to_local(), local_result)
+
+        test_comb(
+            shard0_spec, shard0_spec, shard0_spec,
+            beta=0.0, alpha=0.5
+        )
+
+        test_comb(
+            shard0_spec, shard0_spec, shard0_spec,
+            beta=0.0, alpha=0.5
+        )
+
+        with self.assertRaises(Exception):
+            test_comb(
+                shard1_spec, shard1_spec, shard1_spec,
+                beta=0.8, alpha=0.5
+            )
+
+        with self.assertRaises(Exception):
+            test_comb(
+                shard2_spec, shard2_spec, shard2_spec,
+                beta=0.8, alpha=0.5
+            )
+
+        with self.assertRaises(Exception):
+            test_comb(
+                shard1_spec, shard1_spec, shard1_spec,
+                beta=0.0, alpha=0.5
+            )
+
+        with self.assertRaises(Exception):
+            test_comb(
+                shard2_spec, shard2_spec, shard2_spec,
+                beta=0.0, alpha=0.5
+            )
+
+        # TODO: add more tests - what cases do we want to support?
+        # TODO: test with replicate
+
+    @with_comms
+    def test_sharded_bmm(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        torch.manual_seed(self.rank)
+        input = torch.rand(3, 5, 8, device=self.device_type)
+        mat_2 = torch.rand(3, 8, 6, device=self.device_type)
+        local_result = torch.bmm(input, mat_2)
+
+        def test_placement_comb(
+            placements1: Sequence[Placement],
+            placements2: Sequence[Placement],
+        ) -> None:
+            input_dt = DTensor.from_local(input, device_mesh, placements1)
+            mat_2_dt = DTensor.from_local(mat_2, device_mesh, placements2)
+            new_dt = torch.bmm(input_dt, mat_2_dt)
+            self.assertEqual(new_dt.to_local(), local_result)
+
+
+        assertionErrorFormat = \
+            "Test failed! AssertionError: Result tensors not equal! Spec: {0} {1}."
+        runtimeErrorFormat = \
+            "Test failed! RuntimeError: \"{2}\"! Spec: {0} {1}."
+        shard0_spec = Shard(0)
+        shard1_spec = Shard(1)
+        shard2_spec = Shard(2)
+        shard_specs = [shard0_spec, shard1_spec, shard2_spec]
+        shard_specs_comb = itertools.product(shard_specs, shard_specs)
+        passlist = [
+            [shard0_spec, shard0_spec],
+            [shard2_spec, shard1_spec],
+        ]
+
+        def placementEqual(
```

Comment:
why we need this helper? it should be redundant I think, we can just use `p1 == p2` to check this